src/types.ts

```diff
@@ -1,50 +1,48 @@
-// `null` is used for a hack to manage shared state with SWR
-// https://github.com/vercel/swr/pull/918
-export type fetcherFn<Data> = ((...args: any) => Data | Promise<Data>) | null
+export type fetcherFn<Data> = (...args: any) => Data | Promise<Data>
```

Comment:
`null` is only accepted for `useSWR` and `useSWRInfinite` doesn't accept `null` as the fetcher, so I've moved `null` to the argument of `useSWR`.